app/models/concerns/client_search.rb

```diff
@@ -17,63 +19,64 @@ def self.text_searcher(text, **kwargs, &block)
+        ## search with these other conditions, first check if the term matches external ids. If no matches are
+        ## found, we do an early return with name-search results.
+        # matches_external_ids = where(search_by_external_id(never_cond, text)).any? if ENV['ALPHANUMERIC_HMIS_EXTERNAL_IDS'] && alpha_numeric && respond_to?(:search_by_external_id) && RailsDrivers.loaded.include?(:hmis_external_apis)
+        matches_external_ids = false
+        unless matches_external_ids
+          # short circuit the rest of search. Since no external IDS are found, this seems to be free text and we can just return
+          # name search results
+          return ClientSearchUtil::NameSearch.perform_as_joinable_query(term: text, clients: self) if resolve_for_join_query
 
-        where = nickname_search(where, text)
-        where = metaphone_search(where, :FirstName, text)
-        where = metaphone_search(where, :LastName, text)
+          return ClientSearchUtil::NameSearch.perform(term: text, clients: self, sorted: sorted)
+        end
       end
 
-      where = search_by_external_id(where, text) if alpha_numeric && respond_to?(:search_by_external_id) && RailsDrivers.loaded.include?(:hmis_external_apis)
+      # dummy condition to start the OR chain. This method needs refactoring
+      where ||= never_cond
+      where = search_by_external_id(where, text) if alpha_numeric && respond_to?(:search_by_external_id) && RailsDrivers.loaded.include?(:hmis_external_apis) && HmisExternalApis::AcHmis::Mci.enabled?
 
-      begin
-        # requires a block to calculate which client_ids are acceptable within the search context.
-        # If you are searching custom names, you must include the join to the custom names association in the block
-        client_ids = block.call(where)
-      rescue RangeError
-        return none
-      end
-
-      # WARNING: Any ids added to client_ids below here could be outside of the search scope
-      if numeric
+      results = nil
+      if numeric && term_is_possibly_pk
+        client_ids = self.where(where).pluck(:id)
         source_client_ids = GrdaWarehouse::WarehouseClient.where(destination_id: text).pluck(:source_id)
         if source_client_ids.any?
           # append destination_id
           client_ids << text
           # append any source client ids for that destination
           client_ids += source_client_ids
+          results = where(id: client_ids)
+        else
+          results = where(where)
         end
+      else
+        results = where(where)
       end
-      where(id: client_ids)
+
+      # we aren't dealing with a fuzzy string search here so we can't really rank the results
+      # return NULL as score as match is binary (either it matches an ID or it doesn't)
+      results = results.select(Arel.sql('"Client"."id" AS client_id'), Arel.sql('NULL AS score')) if resolve_for_join_query
```

Comment:
I think this one could be arel'd as:
```suggestion
      results = results.select(c_t[:id].as('client_id'), Arel.sql('NULL AS score')) if resolve_for_join_query
```

The table name quoting when the table case isn't lower case is one of the gotchas with cross-db use.